### PR TITLE
merge bitcoin#17071...#18206: fuzzing harness backports (part 1)

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

File: src/script/interpreter.h [hunk 1]
```diff
@@ -109,11 +110,12 @@ struct PrecomputedTransactionData
 {
     uint256 hashPrevouts, hashSequence, hashOutputs;
     bool m_ready = false;
+    std::vector<CTxOut> m_spent_outputs;
 
     PrecomputedTransactionData() = default;
 
     template <class T>
-    void Init(const T& tx);
+    void Init(const T& tx, std::vector<CTxOut>&& spent_outputs);
```
Comment (UdjinM6) [inline src/script/interpreter.h:118]: 19953: missing changes in `src/validation.cpp`. was that intentional?

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK (just one question, see below)

File: src/script/interpreter.h (hunk 1 above)
Comment (kittywhiskers) [reply src/script/interpreter.h:118]: Yes, attempting to implement them caused crashes and corruptions and the partial was mostly done to resolve conflicts so only a bare-minimum version was implemented, hence the omissions.

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**